app/models/dialog_field_text_box.rb

```diff
@@ -38,10 +38,12 @@ def validate_field_data(dialog_tab, dialog_group)
 
     return "#{dialog_tab.label}/#{dialog_group.label}/#{label} is required" if required? && value.blank?
 
-    case validator_type
-    when 'regex'
-      return "#{dialog_tab.label}/#{dialog_group.label}/#{label} is invalid" unless value.match(/#{validator_rule}/)
-    end
+    # currently only regex is supported
+    rule = validator_rule if validator_type == 'regex'
+    rule ||= "^[0-9]+$" if data_type == "integer"
```

Comment:
@Fryguy Good to know the alternative. In this case it it not possible for the text to contain `\n`  since the input is typed in a text field.